Nakama/TinyJson/JsonParser.cs

```diff
@@ -169,9 +170,17 @@ private static object ParseValue(Type type, string json)
         {
             if (type == typeof(string))
             {
+                var isUnquotedNumber = Regex.Match(json, "^\\d+$").Success;
```

Comment:
Any idea what the performance implications might be here? If I recall correctly, `ParseValue` is called for every value and doing a `Regex.Match` isn't cheap.

Reply:
Happy to change it to a lightweight alternative, though less robust, would be something like:
```
if (json[i] != '"')
```

Reply:
Don't we already know the type of the JSON value ahead of time though? Why doesn't the `IsPrimitive` check in `ParseValue` catch this?

Reply:
Because this is in a situation where you're trying to parse a number as a string. e.g. to a `Dictionary<string, string>`.

The bug is not present if you do something like parse it to a `Dictionary<string, long>`.

Reply:
I see, it's not present if you parse from a pre-known object either, probably.

Reply:
Yep if you deserialize to an object with an actual `long` property it works fine.